src/user/mod.rs

```diff
@@ -378,6 +378,24 @@ pub fn updates(req: &mut Request) -> CargoResult<Response> {
     Ok(req.json(&R{ versions: versions, meta: Meta { more: more } }))
 }
 
+/// Handles the `GET /users/:user_id/stats` route.
+pub fn stats(req: &mut Request) -> CargoResult<Response> {
+    use  diesel::expression::dsl::sum;
+    let name = &req.params()["user_id"].parse::<i32>().ok().unwrap();
+    let conn = req.db_conn()?;
+    
+    let data = crate_downloads::table.filter(crate_downloads::crate_id.eq_any(
+        crate_owners::table.select(crate_owners::crate_id)
+            .filter(crate_owners::owner_id.eq(name))
+        )).select(sum(crate_downloads::downloads)).first::<i64>(&*conn)?; 
```

Comment:
@sgrif I was trying to take the approach that you suggested. But crates.io does not seem to have model for `crate_downloads` and so `inner_join` gives me this - 

```
the trait `diesel::JoinTo<_>` is not implemented for `schema::crate_downloads::table
```

Is there any other approach I can take to make `inner_join` work? Thanks! :)